File: python/conduits/send_message.py
from pwrpy.pwrwallet import Wallet
from dotenv import load_dotenv
import json
import os
load_dotenv()

# Setting up your wallet in the SDK
seed_phrase = os.getenv("SEED_PHRASE")
wallet = Wallet.new(seed_phrase)

def send_message():
    vida_id = 123
    obj = {"message": "please send me pwr"}
    data = json.dumps(obj).encode('utf-8') # Serialize to JSON bytes
    fee_per_byte = wallet.get_rpc().get_fee_per_byte()

    # Sending the VIDA data transaction
    res = wallet.send_vida_data(vida_id, data, fee_per_byte)
    if res.success:
        print(f"Transaction hash: 0x{res.hash.hex()}")
    else:
        print(f"Transaction failed: {res.error}")
send_message()

File: go/conduits/sendMessage.go
package conduits

import (
    "fmt"
    "os"
    "encoding/json"
    "github.com/joho/godotenv"
    "github.com/pwrlabs/pwrgo/wallet"
)

func SendMessage() {
    // Setting up your wallet in the SDK
    godotenv.Load()
    seedPhrase := os.Getenv("SEED_PHRASE")
    wallet, _ := wallet.New(seedPhrase)

    vidaId := 123
    data, _ := json.Marshal(map[string]string{"message": "please send me pwr"})
    feePerByte := wallet.GetRpc().GetFeePerByte()

    // Sending the VIDA data transaction
    tx := wallet.SendVidaData(vidaId, data, feePerByte)

    if tx.Success {
        fmt.Printf("Transaction Hash: %s\n", tx.Hash)
    } else {
        fmt.Printf("Failed to send transaction: %s\n", tx.Error)
    }
}